src/erk/core/claude_settings.py

```
@@ -0,0 +1,102 @@
+
+    Args:
+        repo_root: Path to the repository root
+
+    Returns:
+        Path to {repo_root}/.claude/settings.json
+    """
+    return repo_root / ".claude" / "settings.json"
+
+
+def has_erk_permission(settings: dict) -> bool:
+    """Check if erk permission is configured in Claude settings.
+
+    Args:
+        settings: Parsed Claude settings dictionary
+
+    Returns:
+        True if Bash(erk:*) permission exists in permissions.allow list
+    """
+    permissions = settings.get("permissions", {})
+    allow_list = permissions.get("allow", [])
+    return ERK_PERMISSION in allow_list
+
+
+def add_erk_permission(settings: dict) -> dict:
+    """Return a new settings dict with erk permission added.
+
+    This is a pure function that doesn't modify the input.
+
+    Args:
+        settings: Parsed Claude settings dictionary
+
+    Returns:
+        New settings dict with Bash(erk:*) added to permissions.allow
+    """
+    # Deep copy to avoid mutating input
+    new_settings = json.loads(json.dumps(settings))
+
+    # Ensure permissions.allow exists
+    if "permissions" not in new_settings:
+        new_settings["permissions"] = {}
+    if "allow" not in new_settings["permissions"]:
+        new_settings["permissions"]["allow"] = []
+
+    # Add permission if not present
+    if ERK_PERMISSION not in new_settings["permissions"]["allow"]:
+        new_settings["permissions"]["allow"].append(ERK_PERMISSION)
+
+    return new_settings
+
+
+def read_claude_settings(settings_path: Path) -> dict | None:
+    """Read and parse Claude settings from disk.
+
+    Args:
+        settings_path: Path to settings.json file
+
+    Returns:
+        Parsed settings dict, or None if file doesn't exist or is invalid
+    """
+    if not settings_path.exists():
+        return None
+
+    try:
+        content = settings_path.read_text(encoding="utf-8")
+        return json.loads(content)
+    except (json.JSONDecodeError, OSError):
+        return None
```

just let it fail. let exception bubble